src/Engine/Managers/EntityManager/EntityManager.cpp

```diff
@@ -22,22 +22,26 @@ EntityManager::EntityManager() {
 EntityManager::~EntityManager() {}
 
 Entity* EntityManager::createEntity( const std::string& name ) {
+    Core::Index idx = m_entities.emplace( new Entity( name ) );
+    auto& ent = m_entities[idx];
+    ent->idx = idx;
 
     std::string entityName = name;
-    if ( entityExists( name ) )
+    bool mustRename = false;
+    if ( entityName == "" )
     {
-        LOG( logWARNING ) << "Entity `" << name << "` already exists";
-        entityName = name + "_";
+        Core::StringUtils::stringPrintf( entityName, "Entity_%u", idx.getValue() );
+        mustRename = true;
     }
-    Core::Index idx = m_entities.emplace( new Entity( entityName ) );
-    auto& ent = m_entities[idx];
-    ent->idx = idx;
-
-    if ( name == "" )
+    while ( entityExists( entityName ) )
+    {
+        LOG( logWARNING ) << "Entity `" << entityName << "` already exists";
+        entityName = entityName + "_";

```

Comment:
Why adding several '_' in the name instead of just one number ??? It is easier to read XXX_8_YYY instead of XXX_________YYY

Reply:
well, didn't thought of that, would be nicer